src/app/rosetta/lib/construction.ml

```diff
@@ -679,7 +679,10 @@ module Parse = struct
             let signature_kind : Mina_signature_kind.t =
               if String.equal network_identifier.network "mainnet" then
                 Mainnet
-              else Testnet
+              else if String.equal network_identifier.network "testnet" then
```

Comment:
Do match clauses exist for `String` for better readability here?

Reply:
Ocaml do have matching clauses for `String`. There's actually a commit using that. But since the original one based on develop didn't use the pattern match here so if I change it here it would be incompatible with develop branch that's why I changed it back to use `String.equal` in the merged version.